libpod/container_api.go

```diff
@@ -102,6 +102,8 @@ func (c *Container) Init() (err error) {
 		return errors.Wrapf(err, "unable to generate hostname file for container")
 	}
 
+	// Set the hostname in the env variables
+	c.config.Spec.Process.Env = append(c.config.Spec.Process.Env, fmt.Sprintf("HOSTNAME=%s", c.config.Spec.Hostname))
```

Comment:
This is too high in the file - do it once c.runningSpec has been populated around line 270, and append into c.runningConfig.Process.Env instead

Reply:
there is no c.runningConfig, please advise

Reply:
Sorry, c.runningSpec - my naming sense confuses even me